src/classes/LVL_LevelAssign_BATCH.cls

```diff
@@ -267,35 +284,149 @@ public without sharing class LVL_LevelAssign_BATCH implements Database.Batchable
     }
     
     /*********************************************************************************************************
-    * @description query the Level object, and return a map of Object Name to a Set of Fields to query.
-    * @return map<String, list<String>> The map of Object Name to its Set of Fields
+    * @description Query the Level object and return a list of Ladder records for each SObject type with levels
+    * @return List<Ladder> ladders for each SObject type with levels
     */
-    public static map<String, set<String>> getMapStrObjectToQueryFields() {
-        
-        // get all our levels
-        list<Level__c> listAllLvls = [select Id, Name, Target__c, Source_Field__c, Level_Field__c, 
-            Previous_Level_Field__c, Active__c
-            from Level__c
-            where Active__c = true
-            order by Target__c, Level_Field__c];
-            
-        // create our map
-        map<String, set<String>> mapStrObjectToQueryFields = new map<String, set<String>>();
+    public static List<Ladder> getLadders() {
+        List<Level__c> levels = [
+            SELECT Id, Target__c, Source_Field__c, Level_Field__c,
+                Previous_Level_Field__c, Active__c, LastModifiedDate
+            FROM Level__c
+            WHERE Active__c = true
+            ORDER BY Target__c, Level_Field__c];
+
+        List<Ladder> ladders = new List<Ladder>();
+
+        Map<String,Ladder> ladderByTarget = new Map<String,Ladder>();
+        for (Level__c level : levels) {
+            if (!ladderByTarget.containsKey(level.Target__c)) {
+                ladderByTarget.put(level.Target__c, new Ladder(level.Target__c));
+            }
+            ladderByTarget.get(level.Target__c).addLevel(level);
+        }
         
-        for (Level__c lvl : listAllLvls) {
-            // for each object, we'll add all its source, level, and previous level fields
-            String strObj = lvl.Target__c;
-            Set<String> setFlds = mapStrObjectToQueryFields.get(strObj);
-            if (setFlds == null) {
-                setFlds = new Set<String>();
-                mapStrObjectToQueryFields.put(strObj, setFlds);
+        return ladderByTarget.values();
+    }
+
+    /*********************************************************************************************************
+    * @description Stores all the Level__c records for an SObject
+    */
+    public class Ladder {
```

Comment:
This took me a while to understand. I couldn't figure out why it was called a Ladder, but I see now that there are basically two instances of Ladder (one for Account and one for Contact) and that each contains a collection of Levels. Makes perfect sense now that I've read the code.

Reply:
I'll add a little more to the description to help clarify